### DISCUSSION TITLE
Server Side Transformations

### DISCUSSION THREAD

**Discussion by yesoreyeram** [grafana/grafana-infinity-datasource#614] [Docs]
Server side transformations are introduced in Infinity 2.0. You can perform basic data manipulation in the server side. 

To perform server side transformations, you need to add a query type of **transformations**. Once this added, **this will perform server side transformation over all the previous infinity queries with backend parser**. 

> **Warning**: Infinity Transformations/Server side transformations, are not available neither for non infinity datasources nor infinity queries with non-backend parsers.

## Example

<img width="1792" alt="image" src="https://github.com/yesoreyeram/grafana-infinity-datasource/assets/153843/a3116cff-d5eb-4092-83bb-2ca17ec1f052">
<img width="1789" alt="image" src="https://github.com/yesoreyeram/grafana-infinity-datasource/assets/153843/bf8b5787-e8b2-4847-95a7-544aa2f4848e">

### Limit

Limit transformation limits the number of result items rows in each query. 

### Filter Expression

Filter the results based on column values in each query.

### Computed Column

Appends a new column based on expression

### Summarize / Group by

Group by results based on aggregation and dimension.

## Notes

> **Note**: All these transformations, are done post processing. ( once server responds with data, infinity backend client manipulates the data. ). If your API supports, such transformations on server side, use those instead of this.

**Comment by gdiana** [comment]
Great will try out
